bpf/process/bpf_generic_retkprobe.c

```diff
@@ -121,19 +113,57 @@ BPF_KRETPROBE(generic_retkprobe_event, unsigned long ret)
 #endif
 	}
 
+	e->ret = ret;
+	e->common.size = size;
+	e->common.ktime = ktime_get_ns();
+
+	tail_call(ctx, &retkprobe_calls, TAIL_CALL_FILTER);
+	return 1;
+}
+
+__attribute__((section("kprobe/2"), used)) int
+BPF_KRETPROBE(generic_retkprobe_copy_arg)
+{
```

Comment:
why is this added?

Reply:
Because the code additions pushed the individual programs over the 4096 instruction limit and therefore wouldn't load on a 4.19 kernel. This change simply follows the existing patterns by using a tail call (or rather splitting a program into two with the first tail-calling the second) to get under the instruction count limit.

Reply:
OK, but this needs to be marked and documented in a commit message in the PR so that it can be reviewed properly.

The patch that this change in is has the following diffstat:
```
 bpf/cgroup/bpf_cgroup_events.h                |   2 +-
 bpf/include/api.h                             |   3 +
 bpf/lib/bpf_cgroup.h                          |  20 ++--
 bpf/lib/bpf_helpers.h                         |  11 +-
 bpf/lib/bpf_task.h                            |  27 ++---
 bpf/lib/generic.h                             |   1 +
 bpf/lib/process.h                             |   2 +-
 bpf/libbpf/bpf_core_read.h                    | 382 +++++++++++++++++++++++++++++++++++++++++++++++++++-----------
 bpf/libbpf/bpf_tracing.h                      |   6 +-
 bpf/process/bpf_execve_bprm_commit_creds.c    |   1 +
 bpf/process/bpf_execve_event.c                |  22 ++--
 bpf/process/bpf_exit.c                        |   1 +
 bpf/process/bpf_exit.h                        |   4 +-
 bpf/process/bpf_generic_retkprobe.c           |  63 ++++++++---
 bpf/process/bpf_generic_tracepoint.c          |  16 +--
 bpf/process/bpf_loader.c                      |   7 +-
 bpf/process/bpf_process_event.h               | 140 +++++++++++------------
 bpf/process/data_event.h                      |  55 +++++----
 bpf/process/generic_calls.h                   |   9 +-
 bpf/process/retprobe_map.h                    |  11 +-
 bpf/process/types/basic.h                     | 327 ++++++++++++++++++++++++++++-------------------------
 bpf/process/types/probe_read_kernel_or_user.h | 150 ++++++++++++++++++++++++
 bpf/process/types/skb.h                       |  65 +++++------
 bpf/process/types/sock.h                      |  35 +++---
 pkg/api/tracingapi/client_kprobe.go           |   2 +
 pkg/sensors/tracing/args.go                   |   9 +-
 pkg/sensors/tracing/generickprobe.go          |  12 +-
 pkg/sensors/tracing/generictracepoint.go      |   5 +-
 pkg/sensors/tracing/genericuprobe.go          |   2 +-
 pkg/sensors/tracing/kprobe_test.go            |  23 ++--
 pkg/sensors/tracing/tracepoint_amd64_test.go  |   8 +-
 pkg/sensors/tracing/tracepoint_test.go        |   4 +-
 32 files changed, 956 insertions(+), 469 deletions(-)

```

Which makes it difficult to review. Please split the changes into smaller independent patches so that the changes needed for this are clearly marked and can be reviewed one-by-one.

Reply:
You returned to the review a bit too soon 😢 . I had pushed some changes but hadn't yet divided up the commit into many commits.

Reply:
Ah, apologies for that.
Saw the push and assumed the PR was ready.